tls/s2n_config.c

```diff
@@ -105,6 +107,9 @@ static int s2n_config_init(struct s2n_config *config)
 
     config->client_hello_cb_mode = S2N_CLIENT_HELLO_CB_BLOCKING;
 
+    config->custom_x509_extension_oids = sk_ASN1_OBJECT_new_null();
+    POSIX_ENSURE_REF(config->custom_x509_extension_oids);
```

Comment:
This is probably cheap, but there's not much point in initializing `custom_x509_extension_oids` on every config, given that this feature won't be used most of the time. Might be best to just initialize it on the first call to `s2n_config_add_custom_x509_extension()`.